pkg/controllers/deprovisioning/types.go

```diff
@@ -119,9 +132,9 @@ func NewCandidate(ctx context.Context, kubeClient client.Client, recorder events
 // disruption cost is highest, and it approaches zero as the node ages towards its expiration time.
 func (c *Candidate) lifetimeRemaining(clock clock.Clock) float64 {
 	remaining := 1.0
-	if c.provisioner.Spec.TTLSecondsUntilExpired != nil {
+	if c.nodePool.Spec.Deprovisioning.ExpirationTTL.Duration > 0 {
```

Comment:
What if the policy is Never? Is that enforced in validation?

Reply:
It's filtered in the `shouldDeprovision` function. AFAIK, the `lifetimeRemaining` function has no bearing on whether the candidate gets consolidated.